dag/energy.yml

```diff
@@ -219,4 +229,56 @@ steps:
     - data://garden/eia/2022-07-27/energy_consumption
     - data://garden/ggdc/2020-10-01/ggdc_maddison
     - data://garden/owid/latest/key_indicators
+  # Ember yearly electricity 2022.
+  data://meadow/ember/2022-12-13/yearly_electricity:
+    - snapshot://ember/2022-12-13/yearly_electricity.csv
+  data://garden/ember/2022-12-13/yearly_electricity:
+    - data://meadow/ember/2022-12-13/yearly_electricity
+    - data://garden/owid/latest/key_indicators
+    - data://garden/reference
+  # Ember combined electricity 2022.
+  data://garden/ember/2022-12-13/combined_electricity:
+  - data://garden/ember/2022-08-01/european_electricity_review
+  - data://garden/ember/2022-12-13/yearly_electricity
+  # Energy - Electricity mix (BP & Ember, 2022).
+  data://garden/energy/2022-12-28/electricity_mix:
+    - data://garden/bp/2022-12-28/statistical_review
+    - data://garden/ember/2022-12-13/combined_electricity
+    - data://garden/owid/latest/key_indicators
+  data://grapher/energy/2022-12-28/electricity_mix:
+    - data://garden/energy/2022-12-28/electricity_mix
+  # Energy - Fossil Fuel Production 2022.
+  data://garden/energy/2022-12-28/fossil_fuel_production:
+    - data://garden/bp/2022-12-28/statistical_review
+    - data://garden/shift/2022-07-18/fossil_fuel_production
+    - data://garden/owid/latest/key_indicators
+  data://grapher/energy/2022-12-28/fossil_fuel_production:
+    - data://garden/energy/2022-12-28/fossil_fuel_production
+  # Energy - Primary energy consumption 2022.
+  data://garden/energy/2022-12-28/primary_energy_consumption:
+    - data://garden/bp/2022-12-28/statistical_review
+    - data://garden/eia/2022-07-27/energy_consumption
+    - data://garden/ggdc/2020-10-01/ggdc_maddison
+    - data://garden/owid/latest/key_indicators
+  data://grapher/energy/2022-12-28/primary_energy_consumption:
+    - data://garden/energy/2022-12-28/primary_energy_consumption
+  # Energy - UK historical electricity (2022).
+  data://garden/energy/2022-12-28/uk_historical_electricity:
+    - data://garden/uk_beis/2022-07-28/uk_historical_electricity
+    - data://garden/energy/2022-12-28/electricity_mix
+  data://grapher/energy/2022-12-28/uk_historical_electricity:
+    - data://garden/energy/2022-12-28/uk_historical_electricity
+  # Energy - Global primary energy (2022).
+  data://garden/energy/2022-12-28/global_primary_energy:
+    - data://garden/smil/2017-01-01/global_primary_energy
+    - data://garden/bp/2022-12-28/statistical_review
+  data://grapher/energy/2022-12-28/global_primary_energy:
+    - data://garden/energy/2022-12-28/global_primary_energy
+  # BP - Energy mix 2022.
+  data://garden/bp/2022-12-28/energy_mix:
+    - data://garden/bp/2022-12-28/statistical_review
+    - data://garden/owid/latest/key_indicators
+    - data://garden/wb/2021-07-01/wb_income
+  data://grapher/bp/2022-12-28/energy_mix:
```

Comment:
I might be missing something out, but I see that several `grapher` recipes are not in `archive/energy.yml`, including:

- `data://grapher/bp/2022-12-28/energy_mix`
- `data://grapher/energy/2022-12-28/global_primary_energy`
- `data://grapher/energy/2022-12-28/uk_historical_electricity`
- `data://grapher/energy/2022-12-28/primary_energy_consumption`
- `data://grapher/energy/2022-12-28/fossil_fuel_production`
- `data://grapher/energy/2022-12-28/electricity_mix`

Is this expected or should we put these in the archived dag too?

Reply:
Hi @lucasrodes you are totally right, however those datasets still exist in grapher db. I think the safest order of events would be to:
1. Merge this PR.
2. Archive the unnecessary grapher db datasets.
3. Archive the grapher etl dataset (and then VersionTracker will warn about other unnecessary dependencies to be archived).

Otherwise, we could find an issue during the PR review and need to fall back to the old grapher db dataset (although this is unlikely).
In the future, VersionTracker should have the option to check grapher db as well, but for now this has to be done manually. I'll do all the necessary archiving after merging!

Reply:
makes sense, thanks for clarifying!